paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/commands/WriteIntoPaimonTable.scala

```diff
@@ -245,7 +245,8 @@ object WriteIntoPaimonTable {
               indexFileHandler,
               partition,
               targetBucketRowNumber,
-              (_) => true))
+              (_) => true,
+              buckFilter))
```

Comment:
Due to Spark AQE, it should be dangerous to do this filter.

Reply:
`buckFilter` should only be related to `numSparkPartitions` and `sparkPartitionId`, which is determined at runtime. Even if `loadIndex` loads all buckets, we still need to use `buckFilter` to filter when `assign`.  Can you help provide a case where there will be problems if only the relevant buckets are loaded in advance?

Reply:
My mistake. You are right.